lib/post_revisor.rb

```diff
@@ -405,6 +405,7 @@ def is_last_post?
   def plugin_callbacks
     DiscourseEvent.trigger(:before_edit_post, @post)
     DiscourseEvent.trigger(:validate_post, @post)
+    DiscourseEvent.trigger(:before_edit_topic, @topic, @editor) if topic_changed?
```

Comment:
Somehow it's needed to limit the scope of webhook, `post_number` is not only one condition, but also a `topic_changed?`. I'll need some way to check whether topic is changed. But `PostRevisor` is a huge object, I don't see any spots sending the whole object. So I would only send the single value in the event. Though it looks a little bit suspicious comparing others.